Adjust responsive theme to match current Right To Know theme on wide screens

equivalentideas opened the pull request:
This enables us to activate the 'responsive' theme and get a more small screen and touch friendly design than the current Right To Know theme, _without_ changing the current widescreen design.

This comes from issue: https://github.com/openaustralia/righttoknow/issues/396

This has involved a lot of changes, big and small to get working. There are still subtle layout differences between the current wide screen version and this version—where there is a few pixels difference in spacing in some places. Some of those are unavoidable as far as I can tell (such as #440), but others could be made to match pixel perfect with more work/time—it's just deciding if it's worth it. Examples: there is a few pixels more space between headings in the helps sections, and the sidebar column on the request page is a few pixels thinner and a consistent width.

Apart from any major bugs we discover, really useful feedback would be which small differences are worth fixing and which aren't. There are so many moving parts in this css, some of those small differences take quite a while to track down and fix—though through this process I've gotten a pretty good idea of how most of it works.

Could a few people please test this on their phones/tablets and report bugs onto the Responsive styling milestone (and link back here).

This scss could be refactored to be simpler and make better use of variables etc. Should I make those changes and make it part of this PR?

equivalentideas committed:
add righttoknow custom css to responsive folder
This adds the styles, which break layout in many places.

equivalentideas committed:
add comment headings to clarify structure

equivalentideas committed:
consolidate #footer styles

equivalentideas committed:
move navigation search styles to header section
(refactor only)

equivalentideas committed:
add sass variables for font-family styles

equivalentideas committed:
replace font-family values with new variable

equivalentideas committed:
override responsive_footer partial with righttoknow footer

equivalentideas committed:
add footer class to match alavateli responsive partial

equivalentideas committed:
use alavateli partial for site credits

equivalentideas committed:
Only show twitter link if twitter user configured
matches Alavateli default template

equivalentideas committed:
remove 'after-footer' div with &nbsp;
matches default Alavateli theme, also seems redundant

equivalentideas committed:
replace fixed-width footer with flexible max-width
uses $row-width variable to match #wrapper width

equivalentideas committed:
replace footer width css with responsive mixins
also sets the background colour on the footer
to maintain it's colour through this change

equivalentideas committed:
consolidate footer styles into nested scss
(refactor only)

equivalentideas committed:
move the #oaf info to the bottom for small screens
Like on They Vote For You, the contact information
for this project is higher priority than the links
to other oaf projects

equivalentideas committed:
simplify footer_contact markup

equivalentideas committed:
set the footer contact info left on small screens

equivalentideas committed:
add responsive_topnav partial to override

equivalentideas committed:
change 'View requests' nav link to 'all requests'
comform to current RightToKnow theme, rather
than default theme which links to 'successful requests'

equivalentideas committed:
move header styles above footer styles (refactor only)

equivalentideas committed:
use variable for primary colour

equivalentideas committed:
use varibale font-family for #frontpage_splash h2

equivalentideas committed:
strip back to mostly alavateli search styles
This gets it working nicely on small screens.
Added TODO because there are issues with this
implementation that should be fixed upstream.

equivalentideas committed:
fix indentation (whitespace only)

equivalentideas committed:
remove border from navigation

equivalentideas committed:
make the logo smaller on small screens

equivalentideas committed:
basic styles for responsive menu toggler

equivalentideas committed:
make 'logged in' links not bold

equivalentideas committed:
space and size logged_in_links like RightToKnow

equivalentideas committed:
only adjust logged in link styles on wider screens

equivalentideas committed:
set the font-size so it doesn't effect layout
Move the font-sizing on the logged in links
to a more specific selector so it doesn't impact
the positioning of the whole block

equivalentideas committed:
position logged in links at top on wide screens

equivalentideas committed:
make sass more readable with nesting (refact only)

equivalentideas committed:
move header search to be a child of #banner_inner
This is so that it can be positioned to match the
Right To Know theme.

equivalentideas committed:
use sass variable for lighter blue colour (refactor only)

equivalentideas committed:
remove unnecessary selector

equivalentideas committed:
use the banner_inner element to set header width

equivalentideas committed:
replace search icon with text label

equivalentideas committed:
match header search to RightToKnow theme

equivalentideas committed:
add settings.scss the theme
so we can override the basic variables

equivalentideas committed:
reduce max width to match RightToKnow theme

equivalentideas committed:
add basic styles to match buttons to RightToKnow theme

equivalentideas committed:
use sass nesting to make style clearer refactor only)

equivalentideas committed:
fine tune button sizing to match RightToKnow theme

equivalentideas committed:
decrease base font size to match RightToKnow theme

equivalentideas committed:
return listings vertical sizing to RightToKnow theme

equivalentideas committed:
fix background syntax

equivalentideas committed:
simplify styles by grouping request_listing styles
(refactor only)

equivalentideas committed:
remove redundant color attribute (refactor only)

equivalentideas committed:
stop request listing collapsing on small screens

equivalentideas committed:
stop quote icon overlapping with text

equivalentideas committed:
set basic paragraph spacing to match RightToKnow theme

equivalentideas committed:
layout the 'make request to this authority' button
to match the RightToKnow theme

equivalentideas committed:
comment out old Firefox patch
This is no longer needed

equivalentideas committed:
make the button sizes more consistent

equivalentideas committed:
move basic element style into 'Basic styles' section
to make scss more readble
whitespace only

equivalentideas committed:
refactor basic link styles
to make it simpler and more readable
(refactor only)

equivalentideas committed:
nest link styles with their parent elements
makes styles more readable
(refactor only)

equivalentideas committed:
remove redundant styles
(refactor only)

equivalentideas committed:
match heading & dt styles to Right To Know theme

equivalentideas committed:
match definition list style to Right To Know theme

equivalentideas committed:
match listings heading to Right To Know theme

equivalentideas committed:
match body_list styles to Right To Know theme

equivalentideas committed:
fix the $row-width for the adjusted $base-font-size
expands the wrapper width to match Right To Know theme

equivalentideas committed:
adjust wide screen header styles to match Right To Know theme

equivalentideas committed:
remove redundant element wrapping the logo link

equivalentideas committed:
move button styles to Buttons section (whitespace only)

equivalentideas committed:
group styles for the same selector (refactor only)

equivalentideas committed:
match placement of #topnav to Right To Know theme

equivalentideas committed:
inherit theme styles for horizontal margin of listings
fixes odd indentation of request listings

equivalentideas committed:
match width of request_right to Right To Know theme
on wide screens

equivalentideas committed:
match wide screen layout of request filters to Righ To Know theme

equivalentideas committed:
reset box-sizing of datepicker
the sizing was off because it was presumably
designed in a different box-sizing context.

equivalentideas committed:
match footer horizontal layout to Right To Know theme
change the level at which the font-size is decreased
so that the 'inner wrapper' sizing is related to the
standard font-size.

equivalentideas committed:
increase the listing font-size on smaller screens

equivalentideas committed:
move date picker styles into forms section (whitespace only)

equivalentideas committed:
group all the front page styles together (whitespace only)

equivalentideas committed:
nest #frontpage_splash styles for readability (refactor only)

equivalentideas committed:
reduce specificity of css selectors (refactor only)

equivalentideas committed:
roughly match frontpage splash layout to Right To Know theme

equivalentideas committed:
adjust sizing and spacing of frontpage_start_request

equivalentideas committed:
improve placement and style of menu toggler
on very small screens

equivalentideas committed:
make logged in links stand out more on small screen menu

equivalentideas committed:
match the spacing below header to Right To Know theme

equivalentideas committed:
hard code in the column widths to match Right To Know theme

equivalentideas committed:
adjust spacing of listings to match Right To Know theme

equivalentideas committed:
match layout of authority box to Right To Know theme
on the 'select an authority' step page when making a
request.

equivalentideas committed:
basic horizontal form pattern to match Right To Know theme
on the request form

equivalentideas committed:
add red theme to alert messages

equivalentideas committed:
remove redundant font declaration (refactor only)

equivalentideas committed:
adjust vertical spacing of form text

equivalentideas committed:
add yellow background colour to form header
to match Right To Know theme

equivalentideas committed:
add colour and font style to request header 'note'
to match Right To Know theme

equivalentideas committed:
match size and indentation to match Right To Know theme

equivalentideas committed:
Style notice to match Right To Know theme

equivalentideas committed:
adjust form type sizes to match Right To Know theme

equivalentideas committed:
indent the 'possibly related requests'

equivalentideas committed:
fix the width of note and input in request header

equivalentideas committed:
match middle stript font family/style to Right To Know

equivalentideas committed:
match 'sign in reason' text style to Right To Know

equivalentideas committed:
match horizontal layout of signin form on wide screens
to Right To Know theme

equivalentideas committed:
roughly match signing inputs and helpers to Right To Know theme
on wide screens only

swap to the horizontal label, input pattern on wider screens.

equivalentideas committed:
decrease base line-height on wide screens
to match Right To Know theme

equivalentideas committed:
make basic form input height smaller on wide screens
to match Right To Know theme

equivalentideas committed:
reduce vertical spacing of signup form on wide screens
to match Right To Know theme

equivalentideas committed:
group all the #sign_together styles for reability
(refactor only)

equivalentideas committed:
reduce space in signup form header
to match Right To Know theme

equivalentideas committed:
move 'header right' box further right for wide screens
to match Right To Know theme

equivalentideas committed:
stop spacing of sidebar button pattern collapsing

equivalentideas committed:
increase spacing around request status notice
to match Right To Know theme

equivalentideas committed:
set the width and styles of #describe_state_form
to match Right To Know theme

equivalentideas committed:
fix the height of head banner_content
to prevent the menu from sliding up if the logo
does not load

equivalentideas committed:
stop menu bar getting too thin on small screens

equivalentideas committed:
fix indentation (whitespace only)

equivalentideas committed:
increase h3 size on front page
to match Right To Know theme

equivalentideas committed:
reduce the heading line height on wide screens
to match Right To Know theme

equivalentideas committed:
match widths of front page splash boxes to Right To Know theme
override the percentage widths with pixle values
to get an exact match and remove old stuff

equivalentideas committed:
fine tune vertical spacing of front page splash
to match Right To Know theme

equivalentideas committed:
move logged_in_links few pixels lower
to match Right To Know theme

equivalentideas committed:
keep menu toggler in vertical middle across small screen sizes

equivalentideas committed:
stop logo getting too big on medium screen sizes

equivalentideas committed:
match column top spacing to Right To Know theme

equivalentideas committed:
decrease size of text before body title
to match Right To Know theme

equivalentideas committed:
make follower count bigger
to match to Right To Know theme

equivalentideas committed:
match requests status form layout to Right To Know
relates to #435

equivalentideas committed:
match <hr> style to Right To Know theme
relates to #435

equivalentideas committed:
reduce margin around h3 on wide screens
to match Right To Know theme

equivalentideas committed:
set the size of <small> text to browser defined 'smaller'
To match Right To Know theme

equivalentideas committed:
reduce vertical spacing of status form on wide screens
to match Right To Know theme

equivalentideas committed:
make submit button thinner
to match Right To Know theme

equivalentideas committed:
fix wrapping of radio labels on small screens
fixes #439

equivalentideas committed:
fix the width of the authority search box on wide screens
to match Right To Know theme

equivalentideas committed:
fix the width of basic text input
compensate for change in box-sizing
to match Right To Know theme

equivalentideas committed:
reduce the margin above footer
to match Right To Know theme

equivalentideas committed:
make footer slightly wider than main site wrapper
to match Right To Know theme
this should probably be reverted later
as it just looks messy

equivalentideas committed:
reduce margin above the footer-right list
to match Right To Know theme

equivalentideas committed:
separate the credit and contact items in 'footer right'
to match Right To Know theme

equivalentideas committed:
vertically align 'reason to sign in' <p> to Right To Know theme

equivalentideas committed:
slightly adjust horizontal spacing of 'sign in/sign up' headings
to Right To Know theme

equivalentideas committed:
prevent "- or -" from wrapping
relates to #441

equivalentideas committed:
make text inputs wider to compensate for box-sizing
to match Right To Know theme

equivalentideas committed:
ajust vertical layout of headings in signup/in form
to match Right To Know theme

equivalentideas committed:
horizontally match signin/up form to Right To Know theme

equivalentideas committed:
reduce the line-height of form labels
to match Right To Know theme

equivalentideas committed:
remove padding from request header
to match Right To Know theme

equivalentideas committed:
reduce font size of request body
to match Right To Know theme

equivalentideas committed:
even out padding of request header text
to match Right To Know theme

equivalentideas committed:
reduce width of request header on wide screens
to match Right To Know theme

equivalentideas committed:
make the request help text wider on wide screens
to match Right To Know theme

equivalentideas committed:
match widths of form elements to Right To Know theme
on the request form
fixes #430

equivalentideas committed:
increase space below front page splash
to match Right To Know theme

equivalentideas committed:
match front page examples style to Right To Know theme
fixes #432

equivalentideas committed:
specify text-rendering to match Right To Know
should fix letter-spacing bug in chrome
relates to #437

equivalentideas committed:
make width of error messages flexible on small screens
fixes #445

equivalentideas committed:
reset heading text-rendering to browser default 'auto'
fixes #437

equivalentideas committed:
adjust the layout of form to match Right To Know theme
form to change email of authority

equivalentideas committed:
remove space above form notes

equivalentideas committed:
indent all form notes on wide screens
to match Right To Know theme

equivalentideas committed:
reduce the line height of public body sidebar
to match Right To Know theme

equivalentideas committed:
reduce the width under authority search form
for wide screens only
to match Right To Know theme

equivalentideas committed:
slightly increase space under sidebar heading
to match Right To Know theme

equivalentideas committed:
slightly reduce vertical spacing of body listing results
to match Right To Know theme

equivalentideas committed:
slightly reduce spacing around page heading
to match Right To Know theme

equivalentideas committed:
adjust spacing of request search header
to match Right To Know theme

equivalentideas committed:
adjust spacing around general search form
to Right To Know them

equivalentideas committed:
add more vertical spacing to sign_together form
to match the Right To Know theme
impacts wide screen only

equivalentideas committed:
slight adjustment of spacing around sign_together header
to match Right To Know theme

equivalentideas committed:
decrease space around checkbox in sign in form
to match Right To Know
only apply to wider screens

equivalentideas committed:
increase space above footer
to match Right To Know theme

equivalentideas committed:
adjust layout of front page search
to match Right To Know theme

equivalentideas committed:
make the 'link to correspondence' icon always visible
this provides an afforence for the feature across all input types
The current 0 - 1 opacity on hover was a problem because
there's no obvious way to discover the link on touch devices

fixes #443

equivalentideas committed:
put the native 'transition' property after prefixed versions
this means it will override the vendor prefix versions
as browser support for the property becomes standardised

equivalentideas committed:
add Right To Know theme close popup button style
these styles were removed in favour of a simple
'x' character in the responsive theme.
This change adds the more buttony style back in.

closes #451

equivalentideas committed:
add space beneath search form <p>'s
This matches the Right To Know across
general cases, which can then be overridden in
special cases.

equivalentideas committed:
reduce space at the top of <dl>'s
this is done the match the space to the top of
'help' pages (e.g. http://localhost:3000/help/privacy)
to the Right To Know theme

equivalentideas committed:
increase space below all #right_column_flip <h2>'s
rather than just the ones in #public_body_list .
This produces a better match to Right To Know theme

equivalentideas committed:
fix overlapping text in request form header

equivalentideas committed:
add spacing above <p>'s in <dt>'s
This ensures there is even space between lists,
headings and paragraphs no matter what the default
browser style is.

Relates to #453

henare commented:
Epic! :tada: This is looking really good, nice work :clap: 

> Apart from any major bugs we discover, really useful feedback would be which small differences are worth fixing and which aren't. There are so many moving parts in this css, some of those small differences take quite a while to track down and fix—though through this process I've gotten a pretty good idea of how most of it works.

Apart from the issues I've opened I don't think we should worry about the small differences now that it's almost completely the same.

What do you think @mlandauer?

> Could a few people please test this on their phones/tablets and report bugs onto the Responsive styling milestone (and link back here).

I've done a pretty thorough test and logged all the issues I found against that milestone.

> This scss could be refactored to be simpler and make better use of variables etc. Should I make those changes and make it part of this PR?

I'd vote no - I think it's more important to get this out there and we can refactor later (that's what everyone says and doesn't do it, I know!).

mlandauer commented:
+1 on all counts

equivalentideas commented:
awesome, thanks for that @mlandauer & @henare . I'll fix those issues in https://github.com/openaustralia/righttoknow/milestones/Responsive%20styling and then we can merge.

henare commented:
:metal:

equivalentideas committed:
reduce size of <h4>'s on wide screens
to match Right To Know theme
relates to issue #453

equivalentideas committed:
add padding to flash notice on small screens
fixes #458

equivalentideas committed:
decrease font size of flash notice for small screens
does not effect wide screens

equivalentideas committed:
remove the indentation from annotation preview on small screens
fixes #459

equivalentideas committed:
fix squashing of 'possible requests' list on small screens
fixes #457

equivalentideas committed:
Change blog link in to go directly to OAF site
this is the same change (but on _responsive_topnav.html
rather than normal _topnav.html)
as commit cd2553188aaede2d9f936051850697924cf4adbe
which says:

> We're doing this because the blog feed is broken, see #305. This
> should be reverted when we fix #305 by fixing #311.

fixes #455

equivalentideas committed:
match specific page margin to site standard
on the "Unhappy about your request?" page
fixes #461

equivalentideas committed:
fix radio button squashing bug for Firefox Andriod
The height of the radio button was being set for small
screens, so that the button would vertically sit inline
with the label.

This sets the width and height to the same width,
which fixes a problem with the ratio of the radio icon
in Firefox Andriod. It also sets negative margin on the
label to pull it up, back inline with the radio icon.

fixes #460

equivalentideas committed:
add space between radio buttons on small screens
the radio inputs/labels are too close together on
small screens, which makes them hard to tap accurately

equivalentideas commented:
squashed that last bug https://github.com/openaustralia/righttoknow/issues/460 

Could @henare or @mlandauer merge this in please?

mlandauer commented:
I'll leave the merge to @henare as I haven't touched the codebase in a bit

henare commented:
:shipit: deployed :rocket: 

:boom: awesome work @equivalentideas :metal:

henare commented:
@equivalentideas would you like to write a brief blog post to announce this?

equivalentideas commented:
> @equivalentideas would you like to write a brief blog post to announce this?

That's a good idea, I'm sure some of the other alaveteli theme developers would be interested.

henare commented:
I was just thinking a quick one for RTK requesters but I like the idea of one for Alaveteli theme developers.

Maybe we should just write the brief one for RTK to start?
